src/libcore/iter.rs

```diff
@@ -2044,6 +2125,17 @@ impl<I> Iterator for Take<I> where I: Iterator{
     }
 
     #[inline]
+    fn nth(&mut self, n: usize) -> Option<I::Item> {
+        if self.n == 0 {
+            None
+        } else {
+            let n = cmp::min(self.n-1, n);
```

Comment:
This doesn't seem right. Consider `[0, 1].iter().take(1).nth(1)`, which should return `None`.